src/hash.cr

```diff
@@ -882,6 +882,29 @@ class Hash(K, V)
     true
   end
 
+  # Returns `true` if `self` is subset of *other*.
+  def <(other : Hash)
+    return false if self == other
```

Comment:
`Hash#==` iterates through all entries which is already applied in `#<=`. We can avoid that. If this is just `self.size == other.size` it eliminates the possibility of both being equal. And a subset can't have the same number of elements as the superset. It can't have more elements either, so the return condition should be `self.size >= other.size`.

Reply:
Sub/superset is not about only a size but also the contents, so checking for just `.size` is not enough here.

Reply:
Yes, but a hash can't be a (true) subset of another if it has a bigger (or equal) size. A size comparison is cheap, so it should be used to return early.

Reply:
There's no need to duplicate it, see below:

https://github.com/crystal-lang/crystal/blob/a43fb946f746fa83174eafbfe9863f12249f0a88/src/hash.cr#L876-L877

Reply:
You're right, I'll add it.

Reply:
And replace `return false if self == other` please. It's redundant.

Reply:
Replace it with what? And why it's redundant?

Reply:
`self == other` can't be true when `size < other.size`.

Reply:
You're right, I'm evidently not thinking clearly today, thanks for bearing with me!
I've removed that line in force-pushed commit.